doc/rbd/rbd-mirroring.rst

```diff
@@ -0,0 +1,233 @@
+===============
+ RBD Mirroring
+===============
+
+.. index:: Ceph Block Device; mirroring
+
+RBD images can be asynchronously mirrored between two Ceph clusters.  This
+capability uses the RBD journaling image feature to ensure crash-consistent
+replication between clusters.  Mirroring is configured on a per-pool basis
+within peer clusters and can be configured to automatically mirror all pool
+images within a pool or only a specific subset of images.  Mirroring is
+configured using the ``rbd`` command.  The ``rbd-mirror`` daemon is responsible
+for pulling image updates from the remote, peer cluster and applying them to
+the image within the local cluster.
+
+.. important:: To use use RBD mirroring, you must have two Ceph clusters and
+   the ability for the ``rbd-mirror`` daemon to connect to both clusters
+   simultaneously.
+
+Pool Configuration
+==================
+
+The following procedures demonstrate how perform the basic administrative
+tasks to configure mirroring using the ``rbd`` command.  Mirroring is
+configured on a per-pool basis within the Ceph clusters.
+
+The pool configuration steps should be performed on both peer clusters. These
+procedures assume two clusters, named "local" and "remote", are reachable from
+a single host for clarity.
+
+See the `rbd`_ manpage for additional details of how to connect to different
+Ceph clusters.
+
+Enable Mirroring
+----------------
+
+To enable mirroring on a pool with ``rbd``, specify the ``mirror pool enable``
+command, the pool name, and the mirroring mode::
+
+        rbd mirror pool enable {pool-name} {mode}
+
+The mirroring mode can either be ``pool`` or ``image``:
+
+* **pool**:  When configured in ``pool`` mode, all images in the pool with the
+  journaling feature enabled are mirrored.
+* **image**: When configured in ``image`` mode, mirroring needs to be
+  `explicitly enabled on each image`_.
+
+For example::
+
+        rbd --cluster local mirror pool enable image-pool pool
+        rbd --cluster remote mirror pool enable image-pool pool
+
+Disable Mirroring
+-----------------
+
+To disable mirroring on a pool with ``rbd``, specify the ``mirror pool disable``
+command and the pool name::
+
+        rbd mirror pool disable {pool-name}
+
+When mirroring is disabled on the pool, any images where mirroring was
+explicitly enabled will also be disabled.
```

Comment:
When mirroring is disabled on a pool in this way, mirroring will also be disabled on any images (within the pool) for which mirroring was enabled explicitly.